src/components/TokenList/index.tsx

```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import TokenItem from '../TokenItem'
+import { code2tokenMap, TokenCode } from 'globals'
+
+
+interface TokenListProps {
+  tokens: TokenCode[],
+  balances: {[code in TokenCode]: number }
```

Comment:
this is TS typing for an Object within an Object yeah?

Reply:
balances is an object with keys, that are any of TokenCode values, that is:
```
balances = {
  ETH: 6578,
  GNO: 12345,
  ...
}
```

it is kinda like array comprehension in python